Bottom: a0cf298d3678188097df26b381fefea3a5e47711
Top:    2ffcd435da63642a4f8413559aa6165e613e3bef
Author: Jason Wang <jasowang@redhat.com>
Date:   2016-05-09 14:30:44 +0800

Refresh of f2

---

diff --git a/drivers/net/tun.c b/drivers/net/tun.c
index 24e63c5..732e30d 100644
--- a/drivers/net/tun.c
+++ b/drivers/net/tun.c
@@ -1732,11 +1732,13 @@ static int tun_peek_len(struct socket *sock)
 			ret = 0;
 	} else {
 		struct sock *sk = sock->sk;
+		struct sk_buff *skb;
 		unsigned long flags;
 
 		spin_lock_irqsave(&sk->sk_receive_queue.lock, flags);
-		ret = skb_peek(&sk->sk_receive_queue);
+		skb = skb_peek(&sk->sk_receive_queue);
 		spin_unlock_irqrestore(&sk->sk_receive_queue.lock, flags);
+		ret = skb ? skb->len : 0;
 	}
 
 	tun_put(tun);
